momentGW/pbc/ints.py

```diff
@@ -307,6 +312,82 @@ def transform(self, do_Lpq=None, do_Lpx=True, do_Lia=True):
 
         logger.timer(self, "transform", *cput0)
 
+    def input_integrals(self):
+        """
+        Build CDERI elements using THC integrals imported from a h5py
+        file containing a dictionary of components. Inside the dict, a
+        'collocation_matrix' and a 'coulomb_matrix' must be contained
+        with shapes (MO, aux) and (aux,aux) respectively.
+        """
+        if self.input_path is None:
+            raise ValueError(
+                "A file path containing the THC integrals is needed for the THC implementation"
+            )
+        if "thc" not in self.input_path or "THC" not in self.input_path:
+            raise ValueError("File path must contain 'thc' or 'THC' for THC implementation")
+
+        thc_eri = h5py.File(self.input_path, "r")
+
+        kpts_imp = np.array(thc_eri["kpts"])
+
+        if kpts_imp.shape[0] != len(self.kpts):
+            raise ValueError("Number of kpts imported differs from pyscf")
+        if not np.allclose(kpts_imp, self.kpts._kpts) and not np.allclose(
```

Comment:
let's add an `__eq__` to the `momentGW.pbc.kpts.KPoints` class to handle this entire block - the `wrap_around` should handle the sign too. I can do this

Reply:
I haven't changed this, but if you don't have time to do it, more than happy to do it myself.